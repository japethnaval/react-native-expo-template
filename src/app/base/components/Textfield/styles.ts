import { StyleSheet } from 'react-native'

export const styles = StyleSheet.create({
  base: {
    height: 40,
    marginBottom: 16,
    paddingHorizontal: 8,
    borderWidth: 1,
    borderRadius: 5,
  },
})

export default styles
