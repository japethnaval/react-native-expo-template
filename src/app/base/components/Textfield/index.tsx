import React from 'react'
import { TextInput, TextInputProps } from 'react-native'
import styles from './styles'

export const Textfield: React.FC<TextInputProps> = ({ ...props }) => <TextInput style={styles.base} {...props} />

export default Textfield
