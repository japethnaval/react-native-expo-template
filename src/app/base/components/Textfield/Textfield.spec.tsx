import React from 'react'
import { shallow } from 'enzyme'
import Textfield from './index'

describe('Textfield', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(<Textfield name="name" defaultValue="Useless text" />)
      expect(component).toMatchSnapshot()
    })
  })
})
