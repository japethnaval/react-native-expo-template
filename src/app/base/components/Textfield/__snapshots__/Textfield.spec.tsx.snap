// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textfield Rendering should match to snapshot 1`] = `
<TextInput
  allowFontScaling={true}
  defaultValue="Useless text"
  name="name"
  rejectResponderTermination={true}
  style={
    Object {
      "borderRadius": 5,
      "borderWidth": 1,
      "height": 40,
      "marginBottom": 16,
      "paddingHorizontal": 8,
    }
  }
  underlineColorAndroid="transparent"
/>
`;
